python/taichi/lang/kernel_impl.py

```diff
@@ -696,28 +694,7 @@ def wrapped(*args, **kwargs):
         @functools.wraps(_func)
         def wrapped(*args, **kwargs):
             _taichi_skip_traceback = 1
-            try:
-                return primal(*args, **kwargs)
-            except RuntimeError as e:
-                if str(e).startswith("TypeError: "):
-                    tb = e.__traceback__
-
-                    while tb:
-                        if tb.tb_frame.f_code.co_name == 'taichi_ast_generator':  # pylint: disable=E1101
-                            tb = tb.tb_next
-                            if sys.version_info < (3, 7):
-                                # The traceback object is read-only on Python < 3.7,
-                                # print the traceback and raise
-                                traceback.print_tb(tb,
-                                                   limit=1,
-                                                   file=sys.stderr)
-                                raise TypeError(str(e)[11:]) from None
-                            # Otherwise, modify the traceback object
-                            tb.tb_next = None
-                            raise TypeError(
-                                str(e)[11:]).with_traceback(tb) from None
-                        tb = tb.tb_next
-                raise
```

Comment:
This part is to tailor the traceback to be more human-readable. You can just change L701-L702 to `except TypeError`

Reply:
The traceback inside ASTTransformer is now included in the exception now in https://github.com/taichi-dev/taichi/pull/3637, so IIUC it is not necessary anymore?

Reply:
I tried on this branch, the traceback is a bit long to me. but mostly LGTM.